source/php/Component/MegaMenu/megaMenu.blade.php

```diff
@@ -50,28 +50,46 @@ class="{{$baseClass}}__item {{$baseClass}}__item--parent o-grid-12 o-grid-6@md o
                         {{ $item['description'] }}
                     @endtypography
                 @endif
+
                 {{-- Children --}}
                 @if (!empty($item['children']))
-                    <ul class="{{$baseClass}}__sublist unlist u-margin__top--2">
+                    <ul class="{{$baseClass}}__sublist {{$baseClass}}__sublist--{{ ($childStyle ? 'flex' : 'list') }} unlist u-margin__top--2">
                         @foreach ($item['children'] as $child)
                             <li id="{{$id}}-item-{{$child['id']}}" class="{{$baseClass}}__item {{$baseClass}}__item--child {{ $child['classNames'] }}">
-                                @link([
-                                    'href' => $child['href'],
-                                    'classList' => [
-                                        $baseClass . '__link', 
-                                        $baseClass . '__link--child'
-                                    ]
-                                ])
-                                @if(isset($child['icon']) && !array_diff(['icon', 'size', 'classList'], array_keys($child['icon'])))
-                                    @icon([
-                                        'icon' => $child['icon']['icon'],
-                                        'size' => $child['icon']['size'],
-                                        'classList' => $child['icon']['classList'],
+                                @if(!empty($childStyle))
+                                    @button([
+                                        'text' => $child['label'],
+                                        'style' => $childStyle,
+                                        'color' => $childStyleColor ?? 'primary',
+                                        'shape' => $childStyleShape ?? 'pill',
+                                        'icon' => $child['icon']['icon'] !== "" ? $child['icon']['icon'] : 'chevron_right',
```

Comment:
!empty($child['icon']['icon'] will work in more cases

Reply:
This one is harder to fix. It should be able to be false. But is the 'icon' key  in $child['icon'] always set or can $child['icon'] be false?